pkg/descheduler/strategies/node_affinity.go

```diff
@@ -66,6 +66,30 @@ func removePodsViolatingNodeAffinityCount(ds *options.DeschedulerServer, strateg
 				}
 				evictedPodCount += nodepodCount[node]
 			}
+		case "preferredDuringSchedulingIgnoredDuringExecution":
+			for _, node := range nodes {
+				klog.V(1).Infof("Processing node: %#v\n", node.Name)
+
+				pods, err := podutil.ListEvictablePodsOnNode(ds.Client, node, evictLocalStoragePods)
+				if err != nil {
+					klog.Errorf("failed to get pods from %v: %v", node.Name, err)
+				}
+
+				for _, pod := range pods {
+					if maxPodsToEvict > 0 && nodepodCount[node]+1 > maxPodsToEvict {
+						break
+					}
+					if pod.Spec.Affinity != nil && pod.Spec.Affinity.NodeAffinity != nil && pod.Spec.Affinity.NodeAffinity.PreferredDuringSchedulingIgnoredDuringExecution != nil {
```

Comment:
This line is the only difference between the 2 switch cases. So most of this could be condensed down to a single block, and you only need a switch for this line, correct?